programs/gateway-derive/src/lib.rs

```diff
@@ -77,14 +89,43 @@ pub mod gateway_derive {
 
     Ok(())
   }
+
+  pub fn create_fee(ctx: Context<CreateFee>, amount: u64, mint: Option<Pubkey>) -> Result<()> {
+    ctx.accounts.fee.version = 0;
```

Comment:
I'd expect that instruction to check if a fee already exists and fail the creation otherwise.

Reply:
Otherwise I'd not separate update and create. Was the reasoning to adhere to a previous version?

Reply:
The anchor macros take care of that.